File: VB/dxPaintValue/MyPropertyGridControl.vb
Imports DevExpress.XtraVerticalGrid
Imports DevExpress.XtraVerticalGrid.Data
Imports DevExpress.XtraVerticalGrid.Rows
Imports System.ComponentModel
Imports System.Drawing.Design

Namespace dxPaintValue
    <ToolboxItem(True)> _
    Public Class MyPropertyGridControl
        Inherits PropertyGridControl

        Protected Overrides Function CreatePainterCore(ByVal eventHelper As DevExpress.XtraVerticalGrid.Painters.PaintEventHelper) As DevExpress.XtraVerticalGrid.Painters.VGridPainter
            Return New MyVGridPainter(eventHelper)
        End Function
        Public Function GetUITypeEditor(ByVal context As DescriptorContext) As UITypeEditor
            Return TryCast(context.PropertyDescriptor.GetEditor(GetType(UITypeEditor)), UITypeEditor)
        End Function
        Public Function GetDescriptorContext(ByVal properties As RowProperties) As DescriptorContext
            Return DataModeHelper.GetDescriptorContext(properties.FieldName)
        End Function
    End Class
End Namespace

File: CS/dxPaintValue/MyPropertyGridControl.cs
using DevExpress.XtraVerticalGrid;
using DevExpress.XtraVerticalGrid.Data;
using DevExpress.XtraVerticalGrid.Rows;
using System.ComponentModel;
using System.Drawing.Design;

namespace dxPaintValue {
    [ToolboxItem(true)]
    public class MyPropertyGridControl : PropertyGridControl {
        protected override DevExpress.XtraVerticalGrid.Painters.VGridPainter CreatePainterCore(DevExpress.XtraVerticalGrid.Painters.PaintEventHelper eventHelper) {
            return new MyVGridPainter(eventHelper);
        }
        public UITypeEditor GetUITypeEditor(DescriptorContext context) {
            return context.PropertyDescriptor.GetEditor(typeof(UITypeEditor)) as UITypeEditor;
        }
        public DescriptorContext GetDescriptorContext(RowProperties properties) {
            return DataModeHelper.GetDescriptorContext(properties.FieldName);
        }
    }
}
